### DISCUSSION TITLE
I already have MongoDB can I use that?

### DISCUSSION THREAD

**Discussion by redimongo** [fedify-dev/fedify#58] [Q&A] [answered]
Hi I am trying to get the accept to respond correctly, however running into some issues, and was recommended to use Fedify, however our users are already in our MongoDB database, and we ideally want to utilise that.

Can this be done?

We also use ExpressJS

**Comment by dahlia** [comment, marked as answer]
> Can this be done?

Yes, Fedify is agnostic to databases, so there's no problem using MongoDB with it!

> We also use ExpressJS

And Fedify is also agnostic to web frameworks. However, I don't have any experience with Express.js, so I can't say for sure that it will work well with it.

**Reply by redimongo** [reply]
So out of interest mainly because I am a little confused. 

is this part  kv: new MemoryKvStore(), meant to get something from the Database? or store something in the database? or does it not matter, I can leave that and just store the users profiles in MongoDB and the keys for each account in the mongodb.

Just don't want to run into an issue if I restart the script.

const federation = new Federation<void>({
  kv: new MemoryKvStore(),
  treatHttps: true,
});

**Reply by dahlia** [reply]
That `kv` is just for Fedify's internal cache, so it's okay to leave it and just store the users in your MongoDB.

**Reply by redimongo** [reply]
@dahlia I found out for some reason I could not just use the MemoryKvStore() as everytime I restarted the server it would not remember stuff (note I am using Docker) So I connected it to MongoDB and I was able to get the tutorial working.

I now have to figure out the  creating a Note, and sending it to the followers, as we will be pushing the notes to the database, so I take it we just have to fire a function to get it to send?

**Reply by dahlia** [reply]
Yeah, you need to call `Context.sendActivity()` method.  See also the [relevant section][1] in the docs.

[1]: https://fedify.dev/manual/send
